pkg/apis/clusterlink.net/v1alpha1/import.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 The ClusterLink Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// Import defines a service that is being imported to the local Peer from a remote Peer.
+type Import struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents the attributes of the imported service.
+	Spec ImportSpec `json:"spec"`
+}
+
+// ImportSource represents an addressable exported service.
+type ImportSource struct {
+	// Peer name where the exported service is defined.
+	Peer string `json:"peer"`
+	// ExportName is the name of the exported service.
+	ExportName string `json:"name"`
+	// ExportNamespace is the namespace of the exported service.
```

Comment:
the namespace it requires? if is empty?

Reply:
Since exports are created by namespace owners, they can create overlapping names (since the object's full name is <ns>/<name>). We need to know, explicitly, which export we refer to.